src/libs/SessionUtils.js

```diff
@@ -28,7 +31,34 @@ function isLoggingInAsNewUser(transitionURL, sessionEmail) {
     return linkedEmail !== sessionEmail;
 }
 
-export {
-    // eslint-disable-next-line import/prefer-default-export
-    isLoggingInAsNewUser,
-};
+let loggedInDuringSession;
+
+// To tell if the user logged in during this session we will check the value of session.authToken once when the app's JS inits. When the user logs out
+// we can reset this flag so that it can be updated again.
+Onyx.connect({
+    key: ONYXKEYS.SESSION,
+    callback: (session) => {
+        if (!_.isUndefined(loggedInDuringSession)) {
+            return;
+        }
+
+        if (session && session.authToken) {
+            loggedInDuringSession = false;
+        } else {
+            loggedInDuringSession = true;
+        }
+    },
+});
+
+function resetDidUserLogInDuringSession() {
+    loggedInDuringSession = undefined;
+}
+
+/**
+ * @returns {boolean}
+ */
+function didUserLogInDuringSession() {
```

Comment:
Looking at this again, would it make more sense / be better to have some other system, like "isOnyxDataEmpty"? Maybe not?

Reply:
It's a good question since all sorts of data could be "empty" and any could be used to signify that we have not yet called `OpenApp` yet. On the one hand, it's a bit arbitrary, but on the other hand it is easy to reason about the "when" and "why" we would call OpenApp (post sign in) over ReconnectApp (all other times).

Looking whether Onyx data is entirely empty or not would present a new challenge. This gets pretty complicated, but the gist of it is that [Onyx can initialize with various default states and when cleared we will reapply those default states](https://github.com/Expensify/react-native-onyx/blob/a33f6046e5bbd8ee4ec01a9d7c0660565332d861/lib/Onyx.js#L1050-L1109). So, there really is no way for Onyx data to ever truly be "empty" as far as the App code is concerned.

Reply:
Thanks for the explanation, makes sense 👍